src/kixi/hecuba/minorca.clj

```diff
@@ -28,35 +61,36 @@
   [houses-ids ids-mapping]
   (set/difference houses-ids ids-mapping))
 
-(defn overwrite-mapping [data mapping-file]
+(defn write-to-file [data mapping-file]
   (with-open [out-file (io/writer mapping-file)]
     (csv/write-csv out-file data)))
 
+
+;; Frist step: check if there are new houses to be created
 (defn pre-processing
   "Check if houses are new. If so then create new instances
   in hecuba and add them to the mapping file."
-  [input-file mapping-file project_id base-url username password]
-  (let [input-data (open-input-file input-file)
-        houses-ids (select-identifiers input-data :house_id)
+  [input-data mapping-file project_id base-url username password]
+  (let [houses-ids (select-identifiers input-data :house_id)
         mapping-content (with-open [in-file (io/reader mapping-file)]
-                          (doall (csv/read-csv in-file)))
+                          (vec (csv/read-csv in-file)))
         mapping-data (map #(zipmap (mapv keyword (first mapping-content)) %)
                           (rest mapping-content))
         map-houses-ids (select-identifiers mapping-data :house_id)
         new-houses (look-up houses-ids map-houses-ids)]
     (if (empty? new-houses) ;; No new house
       (println "NO new house")
-      (overwrite-mapping ;; Create new houses + write mapping csv
-       (->> (pmap
-             (fn [house] (vec (cons house
-                                    (api/create-new-entities
-                                     {:project_id project_id :property_code house}
-                                     (str "Device " house)
-                                     base-url username password))))
-             new-houses)
-            (concat mapping-content)
-            vec)
-       mapping-file))))
+      ;; Create new houses + write mapping csv
+      (-> (->> (pmap
+                (fn [house] (vec (cons house
+                                       (api/create-new-entities
+                                        {:project_id project_id :property_code house}
+                                        (str "Device " house)
+                                        base-url username password))))
+                new-houses)
+               (concat mapping-content)
+               vec)
+          (write-to-file mapping-file)))))
```

Comment:
I'd make `write-to-file` have the seq being written in the last position. Plays w/->> better